streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -2438,6 +2440,60 @@ public void shouldConstructAdminMetrics() {
         assertEquals(testMetricName, adminClientMetrics.get(testMetricName).metricName());
     }
 
+    @Test
+    public void shouldCountFailedStreamThread() {
+        verifyFailedStreamThread(false);
+        verifyFailedStreamThread(true);
+    }
```

Comment:
Could you please specify two separate unit tests? One could be named `shouldRecordFailedStreamThread()` and the other `shouldNotRecordFailedStreamThread()`.